src/main/java/org/openmrs/maven/plugins/model/Server.java

```diff
@@ -0,0 +1,132 @@
+        private String nestedInteractiveMode;
+        private String nestedVersion;
+
+        public ServerBuilder setNestedInteractiveMode(String nestedInteractiveMode) {
+            this.nestedInteractiveMode = nestedInteractiveMode;
+            return this;
+        }
+
+        public ServerBuilder setNestedVersion(String nestedVersion) {
+            this.nestedVersion = nestedVersion;
+            return this;
+        }
+
+        public ServerBuilder setNestedServerId(String nestedServerId) {
+            this.nestedServerId = nestedServerId;
+            return this;
+        }
+
+        public ServerBuilder setNestedDbDriver(String nestedDbDriver) {
+            this.nestedDbDriver = nestedDbDriver;
+            return this;
+        }
+
+        public ServerBuilder setNestedDbUri(String nestedDbUri) {
+            this.nestedDbUri = nestedDbUri;
+            return this;
+        }
+
+        public ServerBuilder setNestedDbUser(String nestedDbUser) {
+            this.nestedDbUser = nestedDbUser;
+            return this;
+        }
+
+        public ServerBuilder setNestedDbPassword(String nestedDbPassword) {
+            this.nestedDbPassword = nestedDbPassword;
+            return this;
+        }
+
+        public Server build() {
+            return new Server(nestedServerId, nestedVersion, nestedDbDriver, nestedDbUri, nestedDbUser, nestedDbPassword, nestedInteractiveMode);
+        }
+    }
+
+    public Server() {};
+
+    public Server(String serverId, String version, String dbDriver, String dbUri, String dbUser, String dbPassword, String interactiveMode) {
+        this.serverId = serverId;
+        this.version = version;
+        this.dbDriver = dbDriver;
+        this.dbUri = dbUri;
+        this.dbUser = dbUser;
+        this.dbPassword = dbPassword;
+        this.interactiveMode = interactiveMode;
+    }
+
+    public String getDbPath() {
+        return dbPath;
+    }
+
+    public void setDbPath(String dbPath) {
+        this.dbPath = dbPath;
+    }
+
+    public String getServerId() {
+        return serverId;
+    }
+
+    public void setServerId(String serverId) {
+        this.serverId = serverId;
+    }
+
+    public String getDbDriver() {
+        return dbDriver;
+    }
+
+    public void setDbDriver(String dbDriver) {
+        this.dbDriver = dbDriver;
+    }
+
+    public String getDbUri() {
+        return dbUri;
+    }
+
+    public void setDbUri(String dbUri) {
+        this.dbUri = dbUri;
+    }
+
+    public String getDbUser() {
+        return dbUser;
+    }
+
+    public void setDbUser(String dbUser) {
+        this.dbUser = dbUser;
+    }
+
+    public String getDbPassword() {
+        return dbPassword;
+    }
+
+    public void setDbPassword(String dbPassword) {
+        this.dbPassword = dbPassword;
+    }
+
+    public String getInteractiveMode() { return interactiveMode; }
+
+    public void setInteractiveMode(String interactiveMode) { this.interactiveMode = interactiveMode; }
```

Comment:
Are setters absolutely necessary here? Immutable classes are great to work with... easy to debug and less error prone and should be preferred.